jobs/Backend/Task/src/Mews.ExchangeRateProvider/ExchangeRateProviderPolicyDecorator.cs

```diff
@@ -0,0 +1,40 @@
+using Mews.ExchangeRateProvider.Exceptions;
+using Polly;
+using Polly.Registry;
+
+namespace Mews.ExchangeRateProvider;
+
+/// <summary>
+/// Decorates calls to an <see cref="IExchangeRateProvider"/> with a Polly Policy
+/// </summary>
+public sealed class ExchangeRateProviderPolicyDecorator : IExchangeRateProvider
```

Comment:
This decorates the exchange rate provider implementation with a Polly policy - I would have also considered wrapping this in a decorator to cache results, as they appear to be updated on a fairly infrequent basis according to the remote file headers.